Could not install emacs-plus@28 because tar can't find xz

<!-- To help us debug your issue, please complete these sections: -->

Please make sure to follow these steps (and mark the checkboxes):

- [X] run `brew update` and try to reproduce the issue again
- [X] run `brew doctor`, fix all issues and try to reproduce your issue again
- except, there are dozens of things being complained about, and none of them seem relevant to me
- [X] run `brew config` and `brew doctor` and include their output

## What you were trying to do

brew install emacs-plus@28 --with-native-comp --with-modern-doom3-icon 

## What happened (include command output)

<pre>
$ brew install emacs-plus@28 --with-native-comp --with-modern-doom3-icon 
==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-28.1.tar.xz
Already downloaded: /Users/greg/Library/Caches/Homebrew/downloads/81fae34a5dbd8042af6a70512829e9d4a11e31e7067bf4da5c5bded31f757129--emacs-28.1.tar.xz
==> Installing emacs-plus@28 from d12frosted/emacs-plus
tar (child): xz: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
Error: Failure while executing; `tar --extract --no-same-owner --file /Users/greg/Library/Caches/Homebrew/downloads/81fae34a5dbd8042af6a70512829e9d4a11e31e7067bf4da5c5bded31f757129--emacs-28.1.tar.xz --directory /private/tmp/d20220726-18980-6dggq8` exited with 2. Here's the output:
tar (child): xz: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
$
</pre>

<details>
  <summary>Command output</summary>
  <pre>

$ brew install emacs-plus@28 --with-native-comp --with-modern-doom3-icon 
==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-28.1.tar.xz
Already downloaded: /Users/greg/Library/Caches/Homebrew/downloads/81fae34a5dbd8042af6a70512829e9d4a11e31e7067bf4da5c5bded31f757129--emacs-28.1.tar.xz
==> Installing emacs-plus@28 from d12frosted/emacs-plus
tar (child): xz: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
Error: Failure while executing; `tar --extract --no-same-owner --file /Users/greg/Library/Caches/Homebrew/downloads/81fae34a5dbd8042af6a70512829e9d4a11e31e7067bf4da5c5bded31f757129--emacs-28.1.tar.xz --directory /private/tmp/d20220726-18980-6dggq8` exited with 2. Here's the output:
tar (child): xz: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
(mlenv) gregs-newbook:homebrew-core greg$

  </pre>
</details>

## Output of `brew config`

<pre>
$ brew config
HOMEBREW_VERSION: 3.5.6-47-g1af0bcd
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1af0bcde2010db28c8f2e5d6433dc215b033006d
Last commit: 26 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 32a174609ecf962443c662f05be52c42264a9927
Core tap last commit: 4 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vi
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 13.1.6 build 1316
Git: 2.29.2 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.5-x86_64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: true
</pre>


## Output of `brew doctor`

<pre>
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ilmbase

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libcord.1.dylib
  /usr/local/lib/libdvdcss.2.dylib
  /usr/local/lib/libffi.6.dylib
  /usr/local/lib/libgc.1.dylib
  /usr/local/lib/libgmp.10.dylib
  /usr/local/lib/libguile-2.2.1.dylib
  /usr/local/lib/libltdl.7.dylib
  /usr/local/lib/libnnz11.dylib
  /usr/local/lib/libociei.dylib
  /usr/local/lib/libocijdbc11.dylib
  /usr/local/lib/libunistring.2.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/gc.h
  /usr/local/include/gc/cord.h
  /usr/local/include/gc/cord_pos.h
  /usr/local/include/gc/ec.h
  /usr/local/include/gc/gc.h
  /usr/local/include/gc/gc_backptr.h
  /usr/local/include/gc/gc_config_macros.h
  /usr/local/include/gc/gc_gcj.h
  /usr/local/include/gc/gc_inline.h
  /usr/local/include/gc/gc_mark.h
  /usr/local/include/gc/gc_pthread_redirects.h
  /usr/local/include/gc/gc_tiny_fl.h
  /usr/local/include/gc/gc_typed.h
  /usr/local/include/gc/gc_version.h
  /usr/local/include/gc/javaxfc.h
  /usr/local/include/gc/leak_detector.h
  /usr/local/include/gmp.h
  /usr/local/include/libltdl/lt_dlloader.h
  /usr/local/include/libltdl/lt_error.h
  /usr/local/include/libltdl/lt_system.h
  /usr/local/include/ltdl.h
  /usr/local/include/unicase.h
  /usr/local/include/uniconv.h
  /usr/local/include/unictype.h
  /usr/local/include/unigbrk.h
  /usr/local/include/unilbrk.h
  /usr/local/include/uniname.h
  /usr/local/include/uninorm.h
  /usr/local/include/unistdio.h
  /usr/local/include/unistr.h
  /usr/local/include/unistring/cdefs.h
  /usr/local/include/unistring/iconveh.h
  /usr/local/include/unistring/inline.h
  /usr/local/include/unistring/localcharset.h
  /usr/local/include/unistring/stdbool.h
  /usr/local/include/unistring/stdint.h
  /usr/local/include/unistring/version.h
  /usr/local/include/unistring/woe32dll.h
  /usr/local/include/unitypes.h
  /usr/local/include/uniwbrk.h
  /usr/local/include/uniwidth.h

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libcord.la
  /usr/local/lib/libffi.la
  /usr/local/lib/libgc.la
  /usr/local/lib/libgmp.la
  /usr/local/lib/libguile-2.2.la
  /usr/local/lib/libltdl.la
  /usr/local/lib/libunistring.la

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/bdw-gc.pc
  /usr/local/lib/pkgconfig/guile-2.2.pc
  /usr/local/lib/pkgconfig/libffi.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libffi.a
  /usr/local/lib/libgmp.a
  /usr/local/lib/libguile-2.2.a
  /usr/local/lib/libltdl.a
  /usr/local/lib/libunistring.a

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  libtool
  gmp
  w3m
  ghostscript
  libunistring
  bdw-gc
  emacs
  libdvdcss
  gawk
  autoconf

</pre>

Same problem:

```
$ brew install emacs-plus --with-native-comp --with-modern-doom3-icon
==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-28.1.tar.xz
Already downloaded: /Users/ajl/Library/Caches/Homebrew/downloads/81fae34a5dbd8042af6a70512829e9d4a11e31e7067bf4da5c5bded31f757129--emacs-28.1.tar.xz
==> Installing emacs-plus@28 from d12frosted/emacs-plus
==> Patching
==> Applying fix-window-role.patch
patching file src/nsterm.m
Hunk #1 succeeded at 8583 (offset -185 lines).
==> Applying system-appearance.patch
patching file src/frame.h
patching file src/nsfns.m
Hunk #1 succeeded at 1315 (offset 59 lines).
patching file src/nsterm.m
Hunk #1 succeeded at 1878 (offset -11 lines).
Hunk #2 succeeded at 5389 (offset -6 lines).
Hunk #3 succeeded at 5635 (offset -6 lines).
Hunk #4 succeeded at 5685 (offset -6 lines).
Hunk #5 succeeded at 5911 (offset -6 lines).
Hunk #6 succeeded at 8916 (offset 14 lines).
Hunk #7 succeeded at 10072 (offset 14 lines).
==> Applying fix-MAC_LIBS-inference-on-Intel.patch
patching file configure.ac
Hunk #1 succeeded at 3843 (offset -282 lines).
==> ./autogen.sh
==> ./configure --enable-locallisppath=/opt/homebrew/share/emacs/site-lisp --infodir=/opt/homebrew/Cellar/emacs-plus@28/28.1/share/info/emacs --pre
==> gmake
==> gmake install
Last 15 lines from /Users/ajl/Library/Logs/Homebrew/emacs-plus@28/04.gmake:
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
usage: gzip [-vcD] [-1-9] [file ...]
gmake: *** [Makefile:605: install-arch-indep] Error 1

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
Could not install emacs-plus@28 because tar can't find xz  https://github.com/d12frosted/homebrew-emacs-plus/issues/481
Could not install with ``brew install emacs-plus@29 --with-native-comp --with-elrumo2-icon`` https://github.com/d12frosted/homebrew-emacs-plus/issues/473

$ tar --version
tar (GNU tar) 1.34
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.

$ xz --version
xz (XZ Utils) 5.2.5
liblzma 5.2.5

$ which tar
/opt/homebrew/opt/gnu-tar/libexec/gnubin/tar

$ which xz
/opt/homebrew/bin/xz

$ sw_vers
ProductName:	macOS
ProductVersion:	12.4
BuildVersion:	21F79

$ brew --version
Homebrew 3.5.6
Homebrew/homebrew-core (git revision af6f88f21d5; last commit 2022-07-29)
```

OK I added a dependency on xz to the formula, and it does in fact get past the 'can't find xz' problem. But now I'm getting this problem: 

```
==> Patching
==> Applying fix-window-role.patch
patching file src/nsterm.m
Hunk #1 succeeded at 8583 (offset -185 lines).
==> Applying system-appearance.patch
patching file src/frame.h
patching file src/nsfns.m
Hunk #1 succeeded at 1315 (offset 59 lines).
patching file src/nsterm.m
Hunk #1 succeeded at 1878 (offset -11 lines).
Hunk #2 succeeded at 5389 (offset -6 lines).
Hunk #3 succeeded at 5635 (offset -6 lines).
Hunk #4 succeeded at 5685 (offset -6 lines).
Hunk #5 succeeded at 5911 (offset -6 lines).
Hunk #6 succeeded at 8916 (offset 14 lines).
Hunk #7 succeeded at 10072 (offset 14 lines).
==> ./autogen.sh
==> ./configure --enable-locallisppath=/usr/local/share/emacs/site-lisp --infodir=/usr/local/Cellar/emacs-plus@28/2
==> gmake
Last 15 lines from /Users/greg/Library/Logs/Homebrew/emacs-plus@28/03.gmake:
Loading loadup.el (source)...
Dump mode: pbootstrap
Using load-path (/private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/lisp /private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/lisp/emacs-lisp /private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/lisp/progmodes /private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/lisp/language /private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/lisp/international /private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/lisp/textmodes /private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/lisp/vc)
Loading emacs-lisp/byte-run...
Loading emacs-lisp/backquote...
Loading subr...
Loading version...
Loading widget...
Loading custom...
Loading emacs-lisp/map-ynp...
Loading international/mule...
Loading international/mule-conf...
gmake[1]: *** [Makefile:868: bootstrap-emacs.pdmp] Illegal instruction: 4
gmake[1]: Leaving directory '/private/tmp/emacs-plusA28-20220808-64118-f1dvd7/emacs-28.1/src'
gmake: *** [Makefile:449: src] Error 2

Do not report this issue to Homebrew/brew or Homebrew/core!
``` 

I will look around to see if this is another known problem there's a fix for.

--- a/Formula/emacs-plus@28.rb
+++ b/Formula/emacs-plus@28.rb
@@ -59,6 +59,7 @@ class EmacsPlusAT28 < EmacsBase
     depends_on "gmp" => :build
     depends_on "libjpeg" => :build
     depends_on "zlib" => :build
+    depends_on "xz" => :build
   end
 
   #

--- a/Formula/emacs-plus@29.rb
+++ b/Formula/emacs-plus@29.rb
@@ -52,6 +52,7 @@ class EmacsPlusAT29 < EmacsBase
     depends_on "gmp" => :build
     depends_on "libjpeg" => :build
     depends_on "zlib" => :build
+    depends_on "xz" => :build
   end
 
   #

